### DISCUSSION TITLE
Preparing for 0.7.1 release

### DISCUSSION THREAD

**Discussion by dstebila** [open-quantum-safe/liboqs#1140] [General]
- [x] Discuss: do we need to [update SOVERSION](https://github.com/open-quantum-safe/liboqs/issues/1081)?
- [x] Create release candidate 1; [update version](https://github.com/open-quantum-safe/liboqs/wiki/Release-process)
- [x] Check that constant time tests pass
- [x] Check downstream tests pass
- [x] Check no performance regressions
- [x] Check interop server

**Comment by baentsch** [comment]
Regarding SOVERSION, I tend to argue we could keep it as-is as we only added algorithms but did not change implementations in a breaking way (as would be externally observable, say by KAT changes) nor did we change oqs-common APIs (which triggered #1081 if I recall correctly): Thus, an app SO-linked to liboqs (0.7.0) should keep running if suddenly confronted with an 0.7.1 version of it.

**Comment by dstebila** [comment]
I've run the constant time (generic and haswell) tests for commit 188e6a38b2bbc1795a105e21bc6925ac266df90b for all algorithms and they pass successfully.

**Comment by baentsch** [comment]
Interop server updated and downstream tests (via oqs-curl, oqs-provider docker images) successfully performed. 

Checking against performance regression: ~How do you suggest we do this? Spot checks (algs, platforms, optimization options) or something more rigorous?~ Edit: Disregard the question: I'll add a script combining (& then comparing) `profiling` runs (vs. current `speed_*` runs). Should be available soon...

**Reply by dstebila** [reply]
Okay, sounds good. I was going to eyeball the various profiling outputs and see if it is showing the behaviour we expect, but a script is even better.

**Reply by baentsch** [reply]
> I was going to eyeball the various profiling outputs

Oops -- that reminds me that the profiling image needs rebuilding (just triggered) -> When complete (and run) you truly can check 0.7.1 results....

**Reply by baentsch** [reply]
Code seems to work OK: Here's the result when running `./scripts/noregress.sh 0.7.0 "-DCMAKE_C_COMPILER=gcc10-gcc"` on `aarch64`:

```
Alg: Classic-McEliece-6688128, Op: keygen, Val1: 1.25; Val2: 1.57; diff: -25.60%
Alg: Classic-McEliece-6688128, Op: keygencycles, Val1: 800135885; Val2: 638844006; diff: 20.16%
Alg: Classic-McEliece-6960119, Op: keygencycles, Val1: 654102682; Val2: 765576960; diff: -17.04%
Alg: Kyber512, Op: keygen, Val1: 23534.33; Val2: 47735.67; diff: -102.83%
Alg: Kyber512, Op: keygencycles, Val1: 42422; Val2: 20881; diff: 50.78%
Alg: Kyber512, Op: encaps, Val1: 18592.67; Val2: 40720.33; diff: -119.01%
Alg: Kyber512, Op: encapscycles, Val1: 53716; Val2: 24489; diff: 54.41%
Alg: Kyber512, Op: decaps, Val1: 15724.0; Val2: 46734.33; diff: -197.22%
Alg: Kyber512, Op: decapscycles, Val1: 63531; Val2: 21330; diff: 66.43%
Alg: Kyber768, Op: keygen, Val1: 14422.0; Val2: 32121.33; diff: -122.72%
Alg: Kyber768, Op: keygencycles, Val1: 69271; Val2: 31061; diff: 55.16%
Alg: Kyber768, Op: encaps, Val1: 11994.0; Val2: 26569.33; diff: -121.52%
Alg: Kyber768, Op: encapscycles, Val1: 83308; Val2: 37563; diff: 54.91%
Alg: Kyber768, Op: decaps, Val1: 10339.67; Val2: 29514.33; diff: -185.45%
Alg: Kyber768, Op: decapscycles, Val1: 96648; Val2: 33807; diff: 65.02%
Alg: Kyber1024, Op: keygen, Val1: 9517.33; Val2: 21580.0; diff: -126.74%
Alg: Kyber1024, Op: keygencycles, Val1: 105003; Val2: 46274; diff: 55.93%
Alg: Kyber1024, Op: encaps, Val1: 8300.67; Val2: 18262.0; diff: -120.01%
Alg: Kyber1024, Op: encapscycles, Val1: 120401; Val2: 54693; diff: 54.57%
Alg: Kyber1024, Op: decaps, Val1: 7257.0; Val2: 19727.0; diff: -171.83%
Alg: Kyber1024, Op: decapscycles, Val1: 137734; Val2: 50624; diff: 63.25%
Alg: LightSaber-KEM, Op: keygen, Val1: 30228.67; Val2: 41407.67; diff: -36.98%
Alg: LightSaber-KEM, Op: keygencycles, Val1: 33012; Val2: 24084; diff: 27.04%
Alg: LightSaber-KEM, Op: encaps, Val1: 24348.0; Val2: 36809.0; diff: -51.18%
Alg: LightSaber-KEM, Op: encapscycles, Val1: 41003; Val2: 27100; diff: 33.91%
Alg: LightSaber-KEM, Op: decaps, Val1: 22188.0; Val2: 36778.67; diff: -65.76%
Alg: LightSaber-KEM, Op: decapscycles, Val1: 45003; Val2: 27122; diff: 39.73%
Alg: Saber-KEM, Op: keygen, Val1: 16442.67; Val2: 26152.33; diff: -59.05%
Alg: Saber-KEM, Op: keygencycles, Val1: 60750; Val2: 38170; diff: 37.17%
Alg: Saber-KEM, Op: encaps, Val1: 13549.0; Val2: 23066.33; diff: -70.24%
Alg: Saber-KEM, Op: encapscycles, Val1: 73740; Val2: 43286; diff: 41.30%
Alg: Saber-KEM, Op: decaps, Val1: 12315.33; Val2: 22641.67; diff: -83.85%
Alg: Saber-KEM, Op: decapscycles, Val1: 81137; Val2: 44098; diff: 45.65%
Alg: FireSaber-KEM, Op: keygen, Val1: 10197.0; Val2: 17814.33; diff: -74.70%
Alg: FireSaber-KEM, Op: keygencycles, Val1: 98000; Val2: 56068; diff: 42.79%
Alg: FireSaber-KEM, Op: encaps, Val1: 8628.33; Val2: 15802.0; diff: -83.14%
Alg: FireSaber-KEM, Op: encapscycles, Val1: 115831; Val2: 63217; diff: 45.42%
Alg: FireSaber-KEM, Op: decaps, Val1: 7904.33; Val2: 15407.0; diff: -94.92%
Alg: FireSaber-KEM, Op: decapscycles, Val1: 126450; Val2: 64839; diff: 48.72%
Alg: Rainbow-I-Classic, Op: sign, Val1: 399.4; Val2: 312.69; diff: 21.71%
Alg: Rainbow-I-Classic, Op: signcycles, Val1: 2503611; Val2: 3197517; diff: -27.72%
```
-> One can clearly see the impact of #1131 and #1117. Well done @Martyrshot! The Rainbow and McEliece deviations occur in every nightly run, so are no reason for concern. An `x86_64` test run also showed no "surprising" deviations.

I thus propose merging https://github.com/open-quantum-safe/liboqs/pull/1152 and declaring the RC OK.
